src/eos_gates_windows.js

```diff
@@ -1,28 +1,88 @@
 
 const Lang = imports.lang;
 
+const Flatpak = imports.gi.Flatpak;
 const GLib = imports.gi.GLib;
 const Gio = imports.gi.Gio;
-const EosMetrics = imports.gi.EosMetrics;
+const Gtk = imports.gi.Gtk;
 
 const EosGates = imports.eos_gates;
+EosGates.setupEnvironment();
 
 const WHITELISTED_APPS = [
     // This is an unlikely binary name to run into, so just add it
     // in here for testing purposes...
     { processName: "com.endlessm.test-whitelist.exe" },
 ];
 
+function generateEndlessInstallerEntry() {
+    let entry = {
+        regex: /endless-installer.*.exe/,
+        overrideHelpMessage: _("You are already running Endless OS!")
+    };
+
+    // If we detect the presence of eos-installer.desktop, inform
+        // the user on how to actually install Endless OS.
+    if (!!Gio.DesktopAppInfo.new("eos-installer.desktop")) {
+        entry.overrideHelpMessage = _("You are already running Endless OS from live media!");
+        entry.desktopInfo = { id: 'eos-installer.desktop' },
+        entry.replacementInfo = {
+            appName: _("Endless OS Installer"),
+            description: _("You can install Endless OS by running the Endless OS Installer.")
```

Comment:
You *can*, but I suspect that at least in these special cases the message we'll want to show will be different.